libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/model/MapboxExitProperties.kt

```diff
@@ -0,0 +1,111 @@
+import com.mapbox.api.directions.v5.models.ManeuverModifier
+import com.mapbox.navigation.ui.maneuver.R
+
+/**
+ * A class that allows you to define various properties you wish to use to render for [ExitComponentNode]
+ * of a given [PrimaryManeuver], [SecondaryManeuver] or [SubManeuver].
+ * The priority of fallback is in the order:
+ * - shouldFallbackWithDrawable
+ * - shouldFallbackWithText
+ * If both are set to false, no fallback will be applied and only the exit text associated with the
+ * [Maneuver] will be rendered.
+ *
+ * @param shouldFallbackWithDrawable set to true if you want to use [fallbackDrawable] in case
+ * if the [Maneuver] contains an [ExitComponentNode] with a [ManeuverModifier] value other than left
+ * or right.
+ * @param shouldFallbackWithText set to true if you don't want to use a [fallbackDrawable] but prepend
+ * text "Exit" to the exit number in the [Maneuver]
+ * @param exitBackground background to be set to the text view
+ * @param fallbackDrawable drawable to be used in case [ManeuverModifier] has a different value other
+ * than left or right
+ * @param exitLeftDrawable drawable to be used when [ManeuverModifier] is left
+ * @param exitRightDrawable drawable to be used when [ManeuverModifier] is right
+ */
+sealed class MapboxExitProperties(
+    val shouldFallbackWithText: Boolean,
+    val shouldFallbackWithDrawable: Boolean,
+    @DrawableRes val exitBackground: Int,
+    @DrawableRes val fallbackDrawable: Int,
+    @DrawableRes val exitLeftDrawable: Int,
+    @DrawableRes val exitRightDrawable: Int
+) {
+
+    /**
+     * An implementation of [MapboxExitProperties] that allows you to define various properties you
+     * wish to use to render for [ExitComponentNode] of a given [PrimaryManeuver], [SecondaryManeuver]
+     * or [SubManeuver]. The properties specified in this implementation will be used in countries
+     * that follow MUTCD convention.
+     * The priority of fallback is in the order:
+     * - shouldFallbackWithDrawable
+     * - shouldFallbackWithText
+     * If both are set to false, no fallback will be applied and only the exit text associated with the
+     * [Maneuver] will be rendered.
+     *
+     * @param shouldFallbackWithDrawable set to true if you want to use [fallbackDrawable] in case
+     * if the [Maneuver] contains an [ExitComponentNode] with a [ManeuverModifier] value other than left
+     * or right.
+     * @param shouldFallbackWithText set to true if you don't want to use a [fallbackDrawable] but prepend
+     * text "Exit" to the exit number in the [Maneuver]
+     * @param exitBackground background to be set to the text view
+     * @param fallbackDrawable drawable to be used in case [ManeuverModifier] has a different value other
+     * than left or right
+     * @param exitLeftDrawable drawable to be used when [ManeuverModifier] is left
+     * @param exitRightDrawable drawable to be used when [ManeuverModifier] is right
+     */
+    class PropertiesMutcd(
```

Comment:
I think it's okay to have them as classes since these would be defined once at the start of your application and I don't see a use case as to why would someone be changing these properties at runtime. If this is not true than it makes sense to have a data class so it would expose `copy` function which could be used out of the box.

Reply:
I was wondering if we'd like to take advantage of `equals` / `toString` generation, may be useful to compare properties, maybe not ¯\\\_(ツ)\_/¯